src/Selection.js

```diff
@@ -243,17 +244,42 @@ class Selection {
       return this.emit('reset')
     }
 
-    if(click && inRoot)
-      return this.emit('click', {
+    if(click && inRoot) {
+      return this._handleClickEvent(e);
+    }
+
+    // User drag-clicked in the Selectable area
+    if(!click)
+      return this.emit('select', bounds)
+  }
+
+  _handleClickEvent(e) {
+    const { pageX, pageY, clientX, clientY } = getEventCoordinates(e);
+    const now = new Date().getTime();
+
+    if (this._lastClickData && now - this._lastClickData.timestamp < clickInterval) {
+      // Double click event
+      this._lastClickData = {
```

Comment:
shouldn't this be cleared here? It seems like if you click 3 times really fast you'll get 1 click and 2 dblClick events instead of: click, dblClick, click